diff -Naur pkg-config-0.23/configure.in pkg-config-0.23-haiku/configure.in
--- pkg-config-0.23/configure.in	2008-01-16 22:48:07.055836672 +0000
+++ pkg-config-0.23-haiku/configure.in	2013-04-24 06:55:10.042205184 +0000
@@ -4,7 +4,7 @@
 AM_INIT_AUTOMAKE(pkg-config, 0.23)
 AM_MAINTAINER_MODE
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 
 AM_PROG_LIBTOOL
 
diff -Naur pkg-config-0.23/glib-1.2.10/configure.in pkg-config-0.23-haiku/glib-1.2.10/configure.in
--- pkg-config-0.23/glib-1.2.10/configure.in	2008-01-16 22:52:24.052690944 +0000
+++ pkg-config-0.23-haiku/glib-1.2.10/configure.in	2013-04-24 07:00:15.650903552 +0000
@@ -3,6 +3,9 @@
 dnl ***********************************
 builtin(include, acglib.m4)dnl
 
+# add config macro dir
+AC_CONFIG_MACRO_DIR([m4])
+
 # require autoconf 2.13
 AC_PREREQ(2.13)
 
@@ -63,7 +66,7 @@
 AM_INIT_AUTOMAKE($PACKAGE, $VERSION, no-define)
 
 # Specify a configuration file
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_DEFINE_UNQUOTED(GLIB_MAJOR_VERSION, $GLIB_MAJOR_VERSION)
 AC_DEFINE_UNQUOTED(GLIB_MINOR_VERSION, $GLIB_MINOR_VERSION)
@@ -154,7 +157,6 @@
 
 # Checks for programs.
 AC_PROG_CC
-AM_PROG_CC_STDC
 AC_PROG_INSTALL
 
 changequote(,)dnl
