diff -Naur scim-pinyin-0.5.92/configure.ac scim-pinyin-0.5.92-haiku/configure.ac
--- scim-pinyin-0.5.92/configure.ac	2012-07-15 15:29:59.013369344 +0000
+++ scim-pinyin-0.5.92-haiku/configure.ac	2013-04-23 08:19:52.959709184 +0000
@@ -12,7 +12,7 @@
 
 # Init automake stuff
 AM_INIT_AUTOMAKE
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 
 # Init gettext
 ALL_LINGUAS="zh_CN zh_TW ko ja de fr it nl"
